zipline/utils/events.py

```diff
@@ -372,23 +372,64 @@ def should_trigger(self, dt, env):
         return dt.date() not in env.early_closes
 
 
-class NthTradingDayOfWeek(StatelessRule):
+class TradingDayOfWeekRule(StatelessRule):
```

Comment:
Should this be `Stateful` now (not sure the impact of changing the base class here)?